### PR TITLE
Remove Permissions and ObjectLevel permissions

### PR DESCRIPTION
**Pull request by cmelchior**
Permissions are now managed through MongoDB Realm and Stitch functions, not through dedicated SDK API's. 

This PR remove just remove all current API's and tests for Permissions and Object Level Permissions. Stich functionality will be added later.

### CODE REVIEW

**Comment by cmelchior** [comment]
Ready for review

File: realm/realm-library/src/androidTestObjectServer/java/io/realm/SyncedRealmTests.java [hunk 1]
```diff
@@ -92,6 +93,7 @@ private Realm getFullySyncRealm() {
 
     // Test for https://github.com/realm/realm-java/issues/6619
     @Test
+    @Ignore("Going to be removed anyway")
     public void testUpgragendingOptionalSubscriptionFields() throws IOException {
```
Comment (nhachicha) [inline realm/realm-library/src/androidTestObjectServer/java/io/realm/SyncedRealmTests.java]:
```suggestion
    public void testUpgradingOptionalSubscriptionFields() throws IOException {
```

File: realm/realm-library/src/main/java/io/realm/internal/sync/permissions/ObjectPermissionsModule.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package io.realm.internal.sync.permissions;
+
+import io.realm.annotations.RealmModule;
+import io.realm.sync.Subscription;
+
+/**
+ * Realm model classses that are always part of Query-based Realms
```
Comment (nhachicha) [inline realm/realm-library/src/main/java/io/realm/internal/sync/permissions/ObjectPermissionsModule.java]:
```suggestion
 * Realm model classes that are always part of Query-based Realms
```

File: realm/realm-library/src/main/java/io/realm/internal/sync/permissions/ObjectPermissionsModule.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+ */
+@RealmModule(library = true, classes = {
+        Subscription.class
+})
+public class ObjectPermissionsModule {
```
Comment (nhachicha) [inline realm/realm-library/src/main/java/io/realm/internal/sync/permissions/ObjectPermissionsModule.java]: Are we still keeping this permission model for QBS though? I thought all permission including object-level permission are going to be on the Stitch side as well?

Comment (cmelchior) [reply realm/realm-library/src/main/java/io/realm/internal/sync/permissions/ObjectPermissionsModule.java]: No, I'm going to remove this when I remove subscriptions. I just went with minimal changes when removing the permissions.

**Pull request merged**